### DISCUSSION TITLE
Photoprism having issues importing a large video file

### DISCUSSION THREAD

**Discussion by RafaelCenzano** [photoprism/photoprism#2698] [Q&A]
I'm trying to import a 6GB+ video file. I've been getting issues where photoprism when importing throws an error that I pasted below:
`level=info msg="moving files from 1663024875002"`
`level=error msg="import: /photoprism/import/upload/1663024875002 does not exist"`

I was able to get it to import via adding the file via sftp to the upload folder then importing and also by importing/uploading to an album directly. But then I hit the issue:
`level=warning msg="import: 2022/09/20220910_011821_4546812F.mov exceeds file size limit (6523 / 1000 MB)"`

I followed the instructions from photoprism docs to add this variable PHOTOPRISM_ORIGINALS_LIMIT and it's value -1 to my docker container running on unraid. But this didn't seem to do anything and I have no idea how to fix it/get my file to be imported properly. I also realized upload issues could come from nginx so I repeated my attempts on my local ipaddress and this still occured

**Comment by graciousgrey** [comment]
> I followed the instructions from photoprism docs to add this variable PHOTOPRISM_ORIGINALS_LIMIT and it's value -1 to my docker container running on unraid.

Did you restart the container after changing the variable? If the error log still contains (6523 / 1000 MB) it indicates that the variable has not been updated.

**Comment by RafaelCenzano** [comment]
I have restarted it many times but I still have the error.
<img width="1025" alt="Screen Shot 2022-09-14 at 12 01 20 PM" src="https://user-images.githubusercontent.com/32753063/190204911-9fd6895a-be4c-42ee-afbf-cdee5c447f05.png">

**Reply by lastzero** [reply]
We have no experience with Unraid. When using Docker Compose, it is important to use the "stop" & "up" commands instead of "restart" as described in our docs. Otherwise, the new configuration value will not take effect. Do other configuration changes work so that you can rule out such a problem?

**Reply by RafaelCenzano** [reply]
I have turned it off and on and restarted the server and no luck. I tried again just now as a test and had no luck. Other configurations have worked as I have been able to change the import delay and other things like that. Any chance there is a different way to pass in this variable or a different configuration that could be overriding this?

**Reply by RafaelCenzano** [reply]
It appears that the file still imported in to a folder but doesn't appear on the photoprism web app. I tried importing again and got a duplicate file error and upon inspection, I found the file in the media folder. Not sure if this entails a different error or config?

**Reply by lastzero** [reply]
If it's the same file, that is expected: https://docs.photoprism.app/user-guide/library/duplicates/

**Reply by RafaelCenzano** [reply]
Right, but this is happening to a file that isn't showing up in the webapp, and in the logs said import error too large. The file is there and appears to be processed by photoprism but isn't showing up besides in the media folder.

**Reply by lastzero** [reply]
It may not be visible in the built-in file browser because that operates on the index (the database) and not the underlying file system. You can use WebDAV and/or a regular file manager to check the actual contents of the folders if necessary.

**Reply by RafaelCenzano** [reply]
Is there any way to force-add it to the built-in file browser (database)? I was hoping to have an album with pictures and this video to be able to share to people through a shareable link

**Reply by lastzero** [reply]
Only if that filesize limit gets removed. Please try a large value (for example the size of your disk) instead of -1 or 0 as that didn't seem to work. Might set the default value instead of removing the limit, should be checked. A large value will do in practice as no computer as unlimited storage.

**Reply by RafaelCenzano** [reply]
Unfortunately, no change making it a large number, is there any chance that this "PHOTOPRISM_ORIGINALS_LIMIT" isn't the correct variable name?

**Reply by lastzero** [reply]
There's also a resolution limit, but videos are typically not as large that it matters.

**Reply by RafaelCenzano** [reply]
The file is 4k 60 fps shot on iPhone 12. I have imported other videos before so I'm assuming it's not that

**Reply by graciousgrey** [reply]
I just tested the variable again with a docker-compose setup. The variable itself is working.
What might be confusing is that files above the limit are moved to originals even if they won't be indexed. So if you already tried to import the files so that they are physicaly located within the originals directory, you need to re-index after disabling the limit.

- Importing files with a size above the limit results in the following warning : `...exceeds file size limit (27 / 5 MB)`. The files are moved from import to originals but they won't be indexed.
-  Trying to import the files again after disabling the limit results in `... import: 2018/01/20180121_075536_D97CEC6A.cr2 already exists` . The files are not shown in the PhotoPrism UI
- After re-indexing the originals folder the files get indexed and do show up in PhotoPrism

**Reply by RafaelCenzano** [reply]
This didn't seem to work the issue still appears to be the size limit isn't changing. I set the limit to a very high number but still getting ` file size limit (6523 / 1000 MB)`

**Reply by lastzero** [reply]
Would you be able to open a terminal? Don' t ask me how that works on Unraid, but for advanced users it should be possible.

You could then run the `photoprism config` command, which will display an overview of the configuration keys and values, so you can verify that everything is set correctly. If not, I would check if there is a whitelist or blacklist of allowed variables so that the custom value is not passed.  We have reached a point where guessing is not going to get us anywhere.

**Reply by RafaelCenzano** [reply]
```
|------------------------|--------------------------------------------------------------|
|         Value          |                             Name                             |
|------------------------|--------------------------------------------------------------|
| auth-mode              | password                                                     |
| admin-password         | ************************************************************ |
| public                 | false                                                        |
| log-level              | info                                                         |
| debug                  | false                                                        |
| trace                  | false                                                        |
| config-path            | /photoprism/storage/config                                   |
| options-yaml           | /photoprism/storage/config/options.yml                       |
| defaults-yaml          | /etc/photoprism/defaults.yml                                 |
| settings-yaml          | /photoprism/storage/config/settings.yml                      |
| originals-path         | /photoprism/originals                                        |
| originals-limit        |                                                         1000 |
| resolution-limit       |                                                          100 |
| storage-path           | /photoprism/storage                                          |
| sidecar-path           | /photoprism/storage/sidecar                                  |
| albums-path            | /photoprism/storage/albums                                   |
| backup-path            | /photoprism/storage/backup                                   |
| cache-path             | /photoprism/storage/cache                                    |
| cmd-cache-path         | /photoprism/storage/cache/cmd                                |
| thumb-cache-path       | /photoprism/storage/cache/thumbnails                         |
| import-path            | /photoprism/import                                           |
| assets-path            | /opt/photoprism/assets                                       |
| static-path            | /opt/photoprism/assets/static                                |
| build-path             | /opt/photoprism/assets/static/build                          |
| img-path               | /opt/photoprism/assets/static/img                            |
| templates-path         | /opt/photoprism/assets/templates                             |
| temp-path              | /tmp/photoprism_93052682                                     |
| workers                |                                                            2 |
| wakeup-interval        | 15m0s                                                        |
| auto-index             |                                                          180 |
| auto-import            |                                                          180 |
| read-only              | false                                                        |
| experimental           | false                                                        |
| disable-webdav         | false                                                        |
| disable-settings       | false                                                        |
| disable-places         | false                                                        |
| disable-backups        | false                                                        |
| disable-tensorflow     | false                                                        |
| disable-faces          | false                                                        |
| disable-classification | false                                                        |
| disable-ffmpeg         | false                                                        |
| disable-exiftool       | false                                                        |
| disable-heifconvert    | false                                                        |
| disable-darktable      | false                                                        |
| disable-rawtherapee    | false                                                        |
| disable-sips           | true                                                         |
| disable-raw            | false                                                        |
| raw-presets            | false                                                        |
| exif-bruteforce        | false                                                        |
| detect-nsfw            | false                                                        |
| upload-nsfw            | true                                                         |
| tensorflow-version     | 1.15.2                                                       |
| tensorflow-model-path  | /opt/photoprism/assets/nasnet                                |
| default-locale         | en                                                           |
| default-theme          | default                                                      |
| app-icon               | logo                                                         |
| app-name               | PhotoPrism                                                   |
| app-mode               | standalone                                                   |
| wallpaper-uri          |                                                              |
| cdn-url                | /                                                            |
| site-url               | https://photos.rafaelcenzano.com/                            |
| site-author            |                                                              |
| site-title             | PhotoPrism                                                   |
| site-caption           | Pictures                                                     |
| site-description       |                                                              |
| site-preview           | https://photos.rafaelcenzano.com/static/img/preview.jpg      |
| imprint                | PhotoPrism® needs your                                       |
|                        | support!                                                     |
| imprint-url            | https://photoprism.app/membership                            |
| content-uri            | /api/v1                                                      |
| static-uri             | /static                                                      |
| api-uri                | /api/v1                                                      |
| base-uri               | /                                                            |
| http-host              | 0.0.0.0                                                      |
| http-port              |                                                         2342 |
| http-mode              | release                                                      |
| http-compression       |                                                              |
| database-driver        | mysql                                                        |
| database-server        | mariadb:3306                                                 |
| database-host          | mariadb                                                      |
| database-port          |                                                         3306 |
| database-name          | rafphotoprism                                                |
| database-user          | rafprism                                                     |
| database-password      | ************************************************************ |
| database-conns         |                                                           24 |
| database-conns-idle    |                                                           12 |
| darktable-bin          | /usr/bin/darktable-cli                                       |
| darktable-cache-path   |                                                              |
| darktable-config-path  |                                                              |
| darktable-blacklist    | dng,cr3                                                      |
| rawtherapee-bin        | /usr/bin/rawtherapee-cli                                     |
| rawtherapee-blacklist  |                                                              |
| sips-bin               |                                                              |
| heifconvert-bin        | /usr/local/bin/heif-convert                                  |
| ffmpeg-bin             | /usr/bin/ffmpeg                                              |
| ffmpeg-encoder         | libx264                                                      |
| ffmpeg-bitrate         |                                                           50 |
| exiftool-bin           | /usr/bin/exiftool                                            |
| download-token         | 1vw7wsxg                                                     |
| preview-token          |                                                     93052682 |
| thumb-color            | sRGB                                                         |
| thumb-filter           | lanczos                                                      |
| thumb-size             |                                                         4072 |
| thumb-size-uncached    |                                                         7680 |
| thumb-uncached         | false                                                        |
| jpeg-quality           |                                                           95 |
| jpeg-size              |                                                         7680 |
| face-size              |                                                           50 |
| face-score             |                                                     9.000000 |
| face-overlap           |                                                           42 |
| face-cluster-size      |                                                           80 |
| face-cluster-score     |                                                           15 |
| face-cluster-core      |                                                            4 |
| face-cluster-dist      |                                                     0.640000 |
| face-match-dist        |                                                     0.460000 |
| pid-filename           | /photoprism/storage/photoprism.pid                           |
| log-filename           | /photoprism/storage/photoprism.log                           |
|------------------------|--------------------------------------------------------------|
```
Here is the output from that command

**Reply by lastzero** [reply]
Yeah, it seems that the `originals-limit` is 1000, so it is ignored. Inside the container you could also run `echo $PHOTOPRISM_ORIGINALS_LIMIT` to see if the variable is visible there and what value it has, if any (empty output means it is not set).

**Reply by RafaelCenzano** [reply]
Yep appears to be empty, thanks for helping debug this seems to me to be an issue with docker on unraid so I will move back over there for support.

**Reply by tarunspandit** [reply]
any fix?

**Comment by tarunspandit** [comment]
any fix?

**Reply by priyankub** [reply]
If you copied it from the Photoprism documentation site, then (like me), you might have left a space at the end of the variable which is why photoprism did not pick it up.
Try edit the variable in Unraid, and see if there was a space - that did the trick for me

**Reply by lastzero** [reply]
Could you let us know where that is so we can fix it? Thank you!

**Reply by priyankub** [reply]
Looks like the site is fine, but when someone copies and pastes it in the Unraid UI, the OS likely appends extra characters in the copy paste - in my case a space in the variable itself, which is why it didn't work.
So, just double check if the variable name is correctly entered etc. in Unraid.
Another way is to create a separate env file, and pass it as extra parameters eg `--env-file /mnt/cache_appdata/appdata/env_files/photoprism.env`
That way

**Comment by Richigeht** [comment]
Hi,
I set it to -1, get the output in the container as well, but the video file still doesn't seem to be added. New pictures are added just fine....

**Reply by provinzio** [reply]
I am clueless, too. Checked the config option for -1 but am unable to upload a movie with 1,6 GB. In the Webbrowser, I only see an red Upload failed banner. Nothing mentions about it in the web browser log.

**Reply by priyankub** [reply]
If you are using Unraid, check if the variable in the template has a space appended. If it does, remove it, and it should work

**Reply by provinzio** [reply]
I found my problem. I realized after checking `photoprism config`, some config files and the trace log while running `photoprism --trace up`, that no files arrive after the upload. Soon I realized, that my reverse proxy nginx was the culprit. I disabled the buffering completly with this advanced setting ` 	proxy_request_buffering off;`. This can reduce the performance quite a lot. So feel free to increase the buffer limit instead and check that your nginx instance has enough free disk space for it.

**Reply by lastzero** [reply]
We observe that many of the problems that users report to us are caused by the nginx configuration they are using, see the warning in our documentation:
- https://docs.photoprism.app/getting-started/proxies/nginx/

**Reply by provinzio** [reply]
I ran into the next problem... :/ The large file (20 GB) gets uploaded first into `/tmp/multipart-*`, but the container which hosts the photoprism instance does not have enough free disk space. Instead I wanted to change the temporary path to the disk where I store all my photos. Changing TempPath has no effect on where multipart is saved. Have I overlooked something?

**Reply by lastzero** [reply]
Have you tried another way to transfer files e.g. via CIFS (Samba)?

**Reply by provinzio** [reply]
I haven't tried other methods yet. Thanks for your idea, I'll look into it.

Edit: Regarding the multipart upload... as far as I can see, go has no option to customize the multipart save path by its own (https://github.com/golang/go/blob/go1.15.2/src/mime/multipart/formdata.go#L86). Only option would be to completly change the default temp folder of the system, which might be an idea (https://github.com/golang/go/blob/master/src/os/file.go#L445-L456)? For UNIX systems, one can change the TMPDIR environment variable. For now, I adjusted the environemnt variable TMPDIR for the photoprism service.

**Reply by lastzero** [reply]
While you may not be able to set a custom multipart storage path, you can use the documented environment variables to specify a custom temp path:
- https://docs.photoprism.app/getting-started/config-options/#storage
- https://pkg.go.dev/os#TempDir

**Comment by priyankub** [comment]
I use Nextcloud for all my auto uploads, and have the uploaded folder shared with Photoprism.

**Comment by Richigeht** [comment]
HiI am not using nginx as proxy. Not using one at all. Btw. Caddy would also be a waaaaaaay nicer reverse proxy compared toNginx :-) Mit elektrischen GrüßenRicardo KissingerSent from my iPhoneAm 09.04.2024 um 11:35 schrieb Michael Mayer ***@***.***>:﻿
We observe that many of the problems that users report to us are caused by the nginx configuration they are using, see the warning in our documentation:

https://docs.photoprism.app/getting-started/proxies/nginx/

**Reply by lastzero** [reply]
Caddy is a great piece of software and works well as a proxy too. However, Traefik works best for us in combination with Docker, as it automatically detects and configures the available services.

**Reply by priyankub** [reply]
I too moved over to traefik a couple of years back and it's been great!

**Comment by dannyadair** [comment]
Not sure why I was sitting on an old docker compose file, where I got that from or why I didn't get the memo
but in case someone still experiences this issue and comes to this page:

https://docs.photoprism.app/getting-started/config-options/#storage

```
PHOTOPRISM_ORIGINALS_LIMIT	--originals-limit	1000	maximum size of media files in MB (1-100000; -1 to disable)
...
PHOTOPRISM_UPLOAD_LIMIT	--upload-limit	1000	maximum total size of uploaded files in MB (1-100000; -1 to disable)
```
(You can see them in `docker compose exec photoprism photoprism config` as `originals-limit` and `upload-limit`)

Make sure you set the right one(s) - for uploading through the browser UI I needed both :smile: 
(and don't forget `client_max_body_size` if you're serving through nginx)
